core/src/services/upyun/backend.rs

```diff
@@ -0,0 +1,377 @@
+        debug!("backend build started: {:?}", &self);
+
+        let root = normalize_root(&self.config.root.clone().unwrap_or_default());
+        debug!("backend use root {}", &root);
+
+        // Handle bucket.
+        if self.config.bucket.is_empty() {
+            return Err(Error::new(ErrorKind::ConfigInvalid, "bucket is empty")
+                .with_operation("Builder::build")
+                .with_context("service", Scheme::UpYun));
+        }
+
+        debug!("backend use bucket {}", &self.config.bucket);
+
+        let operator = match &self.config.operator {
+            Some(operator) => Ok(operator.clone()),
+            None => Err(Error::new(ErrorKind::ConfigInvalid, "operator is empty")
+                .with_operation("Builder::build")
+                .with_context("service", Scheme::UpYun)),
+        }?;
+
+        let password = match &self.config.password {
+            Some(password) => Ok(password.clone()),
+            None => Err(Error::new(ErrorKind::ConfigInvalid, "password is empty")
+                .with_operation("Builder::build")
+                .with_context("service", Scheme::UpYun)),
+        }?;
+
+        let client = if let Some(client) = self.http_client.take() {
+            client
+        } else {
+            HttpClient::new().map_err(|err| {
+                err.with_operation("Builder::build")
+                    .with_context("service", Scheme::UpYun)
+            })?
+        };
+
+        let signer = UpYunSigner {
+            operator: operator.clone(),
+            password: password.clone(),
+        };
+
+        Ok(UpYunBackend {
+            core: Arc::new(UpYunCore {
+                root,
+                operator,
+                password,
+                bucket: self.config.bucket.clone(),
+                signer,
+                client,
+            }),
+        })
+    }
+}
+
+/// Backend for upyun services.
+#[derive(Debug, Clone)]
+pub struct UpYunBackend {
+    core: Arc<UpYunCore>,
+}
+
+#[async_trait]
+impl Accessor for UpYunBackend {
+    type Reader = IncomingAsyncBody;
+
+    type BlockingReader = ();
+
+    type Writer = UpYunWriters;
+
+    type BlockingWriter = ();
+
+    type Lister = oio::PageLister<UpYunLister>;
+
+    type BlockingLister = ();
+
+    fn info(&self) -> AccessorInfo {
+        let mut am = AccessorInfo::default();
+        am.set_scheme(Scheme::UpYun)
+            .set_root(&self.core.root)
+            .set_native_capability(Capability {
+                stat: true,
+
+                create_dir: true,
+
+                read: true,
+                read_can_next: true,
+
+                write: true,
+                write_can_empty: true,
+                write_can_multi: true,
+                write_with_cache_control: true,
+                write_with_content_type: true,
+
+                write_multi_min_size: Some(1024 * 1024),
+                write_multi_max_size: Some(50 * 1024 * 1024),
```

Comment:
Seems the max size should also be `1MiB`?

> 文件分块：直接切分二进制文件成小块。分块大小固定为 1M。最后一个分块除外。

Reply:
https://help.upyun.com/knowledge-base/rest_api/#e5b9b6e8a18ce5bc8fe696ade782b9e7bbade4bca0

> 文件分块：数据传输时，把文件切分成小块进行上传，分块大小为1M的整数倍，默认1M，最大50M，最后一个分块例外。

Reply:
Got it.

Reply:
sure